### DISCUSSION TITLE
packet loss detection and timeout retransmission mechanism for theia browser

### DISCUSSION THREAD

**Discussion by lqd1434** [eclipse-theia/theia#13869] [Improvements] [answered]
How to add packet loss detection and timeout retransmission mechanism in the process of front-end and back-end communication of theia?
When the network latency is high, when I click a button and need to get some information from the backend, this information may not be sent to the backend, resulting in no feedback for quite a long time.

**Comment by msujew** [comment, marked as answer]
Hey @lqd1434,

since the network stack of Theia is built on TCP (HTTP/Websocket), the protocol already ensures on a low level that packets arrive. Meaning that there's already a retransmission mechanism in place. I don't think our network infrastructure is capable of exposing this low level information.

The only way to identify that the network connection is bad is by waiting for the socket.io ping to timeout - which takes 30 seconds. In that case the application will go into offline mode which will be indicated to the user in the status bar of the app.

**Reply by lqd1434** [reply]
Thanks for your answer, I understand what you mean, I will think about this question again.
